pkg/scheduler/plugin/priority/interpod_affinity.go

```diff
@@ -0,0 +1,235 @@
+			// increment <pm.counts> for every node in the cluster with the same <term.TopologyKey>
+			// value as that of <existingPod>'s node by the constant <ipa.hardPodAffinityWeight>
+			if ipa.hardPodAffinityWeight > 0 {
+				terms := existingPodAffinity.PodAffinity.RequiredDuringSchedulingIgnoredDuringExecution
+				// TODO: Uncomment this block when implement RequiredDuringSchedulingRequiredDuringExecution.
+				//if len(existingPodAffinity.PodAffinity.RequiredDuringSchedulingRequiredDuringExecution) != 0 {
+				//	terms = append(terms, existingPodAffinity.PodAffinity.RequiredDuringSchedulingRequiredDuringExecution...)
+				//}
+				for _, term := range terms {
+					if err := pm.processTerm(&term, existingPod, pod, existingPodNode, nodes, int64(ipa.hardPodAffinityWeight)); err != nil {
+						return err
+					}
+				}
+			}
+			// For every soft pod affinity term of <existingPod>, if <pod> matches the term,
+			// increment <pm.counts> for every node in the cluster with the same <term.TopologyKey>
+			// value as that of <existingPod>'s node by the term's weight.
+			terms := existingPodAffinity.PodAffinity.PreferredDuringSchedulingIgnoredDuringExecution
+
+			if err := pm.processTerms(terms, existingPod, pod, existingPodNode, nodes, 1); err != nil {
+				return err
+			}
+		}
+		if existingHasAntiAffinityConstraints {
+			// For every soft pod anti-affinity term of <existingPod>, if <pod> matches the term,
+			// decrement <pm.counts> for every node in the cluster with the same <term.TopologyKey>
+			// value as that of <existingPod>'s node by the term's weight.
+			terms := existingPodAffinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution
+			if err := pm.processTerms(terms, existingPod, pod, existingPodNode, nodes, -1); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	if hasAffinityConstraints || hasAntiAffinityConstraints {
+		// We need to process all the pods.
+		for _, existingPod := range nodeInfo.Pods() {
+			if err := processPod(existingPod); err != nil {
+				return 0, framework.NewStatus(framework.Error, err.Error())
+			}
+		}
+	} else {
+		// The pod doesn't have any constraints - we need to check only existing
+		// ones that have some.
+		for _, existingPod := range nodeInfo.PodsWithAffinity() {
+			if err := processPod(existingPod); err != nil {
+				return 0, framework.NewStatus(framework.Error, err.Error())
+			}
+		}
+	}
+
+	return 0, nil
+}
+
+// NormalizeScore .
+func (ipa *InterPodAffinityPlugin) NormalizeScore(pc *framework.PluginContext, pod *v1.Pod, scores framework.NodeScoreList) *framework.Status {
+	var pm *podAffinityPriorityMap
+	// FIXME: initialize pod affinity priority map
+	if pm == nil {
+		return framework.NewStatus(framework.Error, "invalid affnity priority map")
+	}
+
+	// convert the topology key based weights to the node name based weights
+	var maxCount, minCount float64
+
+	for nodeName, score := range scores {
```

Comment:
The score list here is an index to score array, we can't find the corresponding score by node name.

Reply:
isn't `currentScore` the same as `score`?

Reply:
Yeah, you're right. It's a mistake, I'll fix this after the depending PR is merged.